src/Runner.Worker/ActionManifestManager.cs

```diff
@@ -108,8 +122,9 @@ public ActionDefinitionData Load(IExecutionContext executionContext, string mani
                             break;
 
                         case "runs":
-                            actionDefinition.Execution = ConvertRuns(executionContext, templateContext, actionPair.Value);
+                            actionDefinition.Execution = ConvertRuns(executionContext, templateContext, actionPair.Value, actionOutputs);
```

Comment:
Let's store `runs` (i.e. `actionPair.Value`) and then call `ConvertRuns` outside of the for loop. Otherwise this adds a constraint that the order matters. Order shouldn't matter for json (for yaml :shrug: but lets make it not matter).